pkg/server/api/svid/v1/service_test.go

```diff
@@ -921,10 +926,186 @@ func TestServiceBatchNewX509SVID(t *testing.T) {
 	}
 }
 
+func TestNewDownstreamX509CA(t *testing.T) {
+	type downstreamCaTest struct {
+		name           string
+		err            string
+		failSigning    bool
+		failDataStore  bool
+		rateLimiterErr error
+		entry          *types.Entry
+		csr            []byte
+		csrTemplate    *x509.CertificateRequest
+		code           codes.Code
+		fetcherErr     string
+	}
+
+	downstreamEntry1 := &types.Entry{
+		Id:         "downstreamCA1",
+		ParentId:   api.ProtoFromID(agentID),
+		SpiffeId:   &types.SPIFFEID{TrustDomain: "example.org", Path: ""},
+		Downstream: true,
+	}
+
+	downstreamEntry2 := &types.Entry{
+		Id:         "downstreamCA2",
+		ParentId:   api.ProtoFromID(agentID),
+		SpiffeId:   &types.SPIFFEID{TrustDomain: "example2.org", Path: ""},
+		Downstream: false,
+	}
+
+	test := setupServiceTest(t)
+	defer test.Cleanup()
+
+	x509CA := test.ca.X509CA()
+
+	for _, tt := range []downstreamCaTest{
+		{
+			name:           "Malformed CSR",
+			rateLimiterErr: nil,
+			err:            "malformed CSR: asn1: structure error",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.InvalidArgument,
+			fetcherErr:     "",
+			entry:          downstreamEntry1,
+		},
+		{
+			name:           "Rate Limiter Err",
+			rateLimiterErr: status.Error(codes.Internal, "rate limit error"),
+			err:            "rate limit error",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.Internal,
+			fetcherErr:     "",
+			entry:          downstreamEntry1,
+		},
+		{
+			name:           "Unauthorized",
+			rateLimiterErr: nil,
+			err:            "error not found or not authorized",
+			failSigning:    false,
+			csr:            []byte{1, 2, 3},
+			code:           codes.NotFound,
+			fetcherErr:     "",
+			entry:          nil,
+		},
+		{
+			name:           "FailDataStore",
+			rateLimiterErr: nil,
+			err:            "bundle not found",
+			failSigning:    false,
+			csrTemplate:    &x509.CertificateRequest{},
+			code:           codes.Internal,
+			fetcherErr:     "",
+			entry:          downstreamEntry1,
+			failDataStore:  true,
+		},
+		{
+			name:           "SuccessfulCARequest",
+			rateLimiterErr: nil,
+			err:            "",
+			failSigning:    false,
+			failDataStore:  false,
+			csrTemplate: &x509.CertificateRequest{
+				// Right now, the request will succeed even if URIs is empty. This is
+				// probably not the right behavior.
+				URIs: []*url.URL{workloadID.URL()},
```

Comment:
I think this comment can be removed.... the URI is only used to convey the public key. The SPIFFE ID is determined by the server.